packages/sdk/src/referral.ts

```diff
@@ -740,49 +790,72 @@ export class ReferralProvider {
       }),
     );
 
-    const txs: VersionedTransaction[] = [];
-    const chunkedInstructions = chunk(claimInstructionParams, 5);
+    const chunkedInstructions = chunk(claimInstructionParams, 4);
 
-    chunkedInstructions.forEach(async (chunkParams) => {
-      let instructions: TransactionInstruction[] = [];
+    const txs: VersionedTransaction[] = await Promise.all(
+      chunkedInstructions.map(async (chunkParams) => {
+        let instructions: TransactionInstruction[] = [];
 
-      for (const {
-        referralTokenAccountPubKey,
-        projectAdminTokenAccount,
-        partnerTokenAccount,
-        mint,
-        preInstructions,
-        tokenProgramId,
-      } of chunkParams) {
-        const tx = await this.program.methods
-          .claim()
-          .accounts({
+        await Promise.all(
+          chunkParams.map(
+            async ({
+              referralTokenAccountPubKey,
+              projectAdminTokenAccount,
+              partnerTokenAccount,
+              mint,
+              preInstructions,
+              tokenProgramId,
+            }) => {
+              const tx = await this.program.methods
+                .claim()
+                .accounts({
+                  payer: payerPubKey,
+                  project: referralAccount.project,
+                  admin: project.admin,
+                  projectAdminTokenAccount,
+                  referralAccount: referralAccountPubKey,
+                  referralTokenAccount: referralTokenAccountPubKey,
+                  partner: referralAccount.partner,
+                  partnerTokenAccount: partnerTokenAccount,
+                  mint,
+                  tokenProgram: tokenProgramId,
+                })
+                .preInstructions(preInstructions)
+                .transaction();
+
+              instructions.push(...tx.instructions);
+            },
+          ),
+        );
+
+        // Priority Fee Instructions
+        const { units, microLamports } =
+          await feeService.getOptimalComputeUnitLimitAndPrice({
+            instructions,
             payer: payerPubKey,
-            project: referralAccount.project,
-            admin: project.admin,
-            projectAdminTokenAccount,
-            referralAccount: referralAccountPubKey,
-            referralTokenAccount: referralTokenAccountPubKey,
-            partner: referralAccount.partner,
-            partnerTokenAccount: partnerTokenAccount,
-            mint,
-            tokenProgram: tokenProgramId,
-          })
-          .preInstructions(preInstructions)
-          .transaction();
-        instructions.push(...tx.instructions);
-      }
+            lookupTables: [lookupTableAccount],
+          });
+        instructions.unshift(
+          ComputeBudgetProgram.setComputeUnitPrice({ microLamports }),
+        );
+        if (units) {
+          instructions.unshift(
+            ComputeBudgetProgram.setComputeUnitLimit({ units }),
+          );
```

Comment:
why `unshift` ?

Reply:
Because want to make the instructions order more clearly defined I guess.
1 -> setComputeUnitLimit
2 -> setComputeUnintPrice

I referring from here
https://solana.com/developers/guides/advanced/how-to-request-optimal-compute#how-to-request-compute-budget

Reply:
ok ok, currently our implementation is push. but i think it's fine